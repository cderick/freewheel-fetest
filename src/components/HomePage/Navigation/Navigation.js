import React from 'react'
import PropTypes from 'prop-types'
import s from './Navigation.module.scss'

class Navigation extends React.Component {
	render() {
		if (!this.props.navItems) return null
		const { 
			navigationIcon,
			navigation
		} = this.props.navItems
		return (
			<section>
				<nav className={s.topnav} id={navigationIcon.id}>
					<a className={`${navigationIcon.classes && navigationIcon.classes}`} id={navigationIcon.id} href={navigationIcon.link} rel="noopener noreferrer" target="_blank" >
						<img className={s.imgResize} src={navigationIcon.image} alt={navigationIcon.alt}/>
					</a>
					{navigation && navigation.length
					&& navigation.map((nav, navInd) => (
						<button key={`nav${navInd}`} onClick={() => this.props.changeNav(nav.id)}>{nav.title}</button>
					))}
				</nav>
			</section>
		)
	}
}

Navigation.propTypes = {
	navItems: PropTypes.objectOf(PropTypes.any),
	changeNav: PropTypes.func,
}

Navigation.defaultProps = {
	navItems: undefined,
}

export default Navigation
