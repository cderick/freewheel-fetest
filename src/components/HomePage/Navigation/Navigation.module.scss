$font_family_sans: sans-serif;

.topnav {
  background-color: #070431;
  overflow: hidden;
  position: static;
  width: 100%;
  top: 0;
  z-index: 999999;

  button {
    outline: 0;
    background: transparent;
    border: none;
    cursor: pointer;
    float: right;
    display: block;
    color: #fff;
    text-align: center;
    padding: 41px;
    text-decoration: none;
    font-size: 17px;
    font-family: $font_family_sans;

    &:hover {
      background-color: #000;
    }
  }

  .imgResize {
    width: 200px;
    padding: 20px 0;
    margin-left: 30px;
    height: auto;
  }

  a {
    outline: 0;
    background: transparent;
    border: none;
    cursor: pointer;
    float: right;
    display: block;
    color: #000;
    text-align: center;
    padding: 15px;
    text-decoration: none;
  }
}
