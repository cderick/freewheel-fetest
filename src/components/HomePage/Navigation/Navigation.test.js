import React from 'react'
import renderer from 'react-test-renderer'
import Navigation from './Navigation'

const componentNoprops = renderer.create(<Navigation />)

describe('Test on <Navigation /> component', () => {
   test('should not render with no props', () => {
      const treeNoprops = componentNoprops.toJSON()
      expect(treeNoprops).toMatchSnapshot()
   })
})