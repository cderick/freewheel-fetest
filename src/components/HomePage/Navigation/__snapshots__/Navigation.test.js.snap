// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <Navigation /> component should not render with no props 1`] = `null`;
